components/Form.js

```diff
@@ -0,0 +1,249 @@
+import styled, { css } from "styled-components";
+import { StyledButton, ButtonGroup } from "@/components/StyledElements/Buttons";
+import { RadioButton } from "@/components/StyledElements/RadioButton";
+import { RangeInput } from "@/components/StyledElements/RangeSlider";
+import { CustomSelect } from "@/components/StyledElements/Select";
+import { useState } from "react";
+
+const FormContainer = styled.form`
+  display: flex;
+  flex-direction: column;
+  max-width: 600px;
+  gap: 0.5rem;
+  padding: 1rem;
+`;
+
+const labelStyles = css`
+  font-weight: bold;
+  margin-top: 0.4rem;
+`;
+const Label = styled.label(labelStyles);
+
+const FieldsetLabel = styled.legend(labelStyles);
+
+const Input = styled.input`
+  padding: 0.5rem;
+  border: 2px solid black;
+  border-radius: 0.5rem;
+`;
+
+const Fieldset = styled.fieldset`
+  padding: 0;
+  border: none;
+`;
+
+const RadioButtonGroup = styled.div`
+  display: flex;
+  gap: 5rem;
+`;
+const RadioButtonLabel = styled.label`
+  display: flex;
+  align-items: flex-end;
+`;
+
+const RangeInputLabels = styled.div`
+  display: flex;
+  justify-content: space-between;
+`;
+
+// Values for custom select components used in form
+const cropTypes = ["Fruit", "Herb", "Vegetable", "Other"];
+const placements = ["Bed", "Pot", "Pot or Bed"];
+const growingConditions = ["Sunny", "Partial Shade"];
+
+export default function Form({ onSubmit, onDismiss }) {
+  // Select Crop Type
+
+  const [currentCropType, setCurrentCropType] = useState(null);
+
+  function handleCropTypeChange(value) {
+    setCurrentCropType(value);
+  }
+
+  // Select Placement
+
+  const [currentPlacement, setCurrentPlacement] = useState(null);
+
+  function handlePlacementChange(value) {
+    setCurrentPlacement(value);
+  }
+
+  // Select Growing Conditions
+
+  const [currentGrowingConditions, setCurrentGrowingConditions] =
+    useState(null);
+
+  function handleGrowingConditionsChange(value) {
+    setCurrentGrowingConditions(value);
+  }
+
+  function checkSelectInput(input, name) {
+    if (!input) {
+      return `Select the preferred ${name} ↓`;
+    } else {
+      return input;
+    }
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    // Handle custom selects: if not null, add data to form data, otherwise show a custom alert message
+    if (
+      currentCropType === null ||
+      currentPlacement === null ||
+      currentGrowingConditions === null
+    ) {
+      alert("Please fill out all inputs");
```

Comment:
You dont have to write that, you can just use `required` but if you want it to be a `alert` its also fine :)